### ISSUE TITLE
password reset form does not work when submitted with keyboard's Enter key

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica#10669]
When you are resetting your password, if you try to submit the form by hitting Enter on your keyboard, you see a JavaScript alert saying "Missing email" (it's a spurious error message; the problem has nothing to do with the presence or absence of an email address). 

If you submit the form by clicking the submit button with your mouse, the form works and your password is reset.

Steps to reproduce:
1. Log out.
2. Use the forgot password link on the login page and enter your email address and submit the form.
3. In the email, click on the reset password link.
4. Type your password once (it can be the same one you used before).
5. Type your password a second time and then immediately hit the Enter key on your keyboard.
6. Observe the incorrect behaviour - see screenshot below.
7. Click the "Set New Password" button with your mouse.
8. Observe the correct behaviour (an alert saying "Your password was successfully changed ...").

The desired behaviour is that step 5 should produce the same result as step 7 (i.e., both methods of submitting the form should set the new password).

![image](https://user-images.githubusercontent.com/1495809/45262578-c03cc880-b45c-11e8-93c2-8a38ca840607.png)

The code that's probably relevent is below, although everything from here on is speculation because I'm not strong on Vue.

https://github.com/HabitRPG/habitica/blob/2ce2100f89bb9eef616fb1ebd6bb10cda75009ad/website/client/components/auth/registerLoginReset.vue#L81-L84
The @click handler allows the `resetPasswordSetNewOneLink` function to be called - this probably what allows step 7 above to work.

https://github.com/HabitRPG/habitica/blob/2ce2100f89bb9eef616fb1ebd6bb10cda75009ad/website/client/components/auth/registerLoginReset.vue#L458-L472
I think this `handleSubmit` function is what goes wrong when you try to submit the form by hitting Enter. The problem is probably that `this.forgotPassword` happens to be true when you're using the Reset Password form, and so `this.forgotPasswordLink();` is executed - that's this code:
https://github.com/HabitRPG/habitica/blob/2ce2100f89bb9eef616fb1ebd6bb10cda75009ad/website/client/components/auth/registerLoginReset.vue#L476-L479
That's the code that's executed in step 2 above. It works in step 2 as it's supposed to, but it fails in step 5 because the Reset Password form does not have a field for the username. I.e., the `forgotPasswordLink` function is not supposed to be executed in step 5.

**Comment by Alys** [comment]
tl;dr for those staff and contractors who do admin email support and maybe for Socialites who might see reports of this occasionally (e.g., from people who have two accounts and can still use Habitica with one of them):

When you see someone report that the Reset Password form tells them "Missing Email" and doesn't let them set a new password, tell them to submit the form by clicking the "Set New Password" button with their mouse. The form won't work if they try to submit it any other way.

**Comment by legitmaxwu** [comment]
could I take a look at this?

**Comment by paglias** [comment]
Sure @legitmaxwu !

**Comment by beffymaroo** [comment]
@legitmaxwu Are you still working on this issue? When you have a moment please check in and let us know. Thanks!

**Comment by legitmaxwu** [comment]
Hello, I am still working on it, but since I am in school, I have not been able to devote much time. Thanks for checking in! If anyone wants to take this instead of me, feel free.

**Comment by wiktordaniec** [comment]
I see the problem still occurs. May I work on this issue?

**Comment by paglias** [comment]
@andrzejZdobywca sure, go ahead!
